requirements-test-py38.txt

```diff
@@ -537,6 +537,10 @@ werkzeug==2.0.2 \
     #   -r requirements-py38.txt
     #   flask
 
-# WARNING: The following packages were not pinned, but pip requires them to be
-# pinned when the requirements file includes hashes. Consider using the --allow-unsafe flag.
-# setuptools
+# The following packages are considered to be unsafe in a requirements file:
```

Comment:
Let's dig deep why and not add --allow-unsafe

Reply:
Without `--allow-unsafe` pip gives me this error while building:

```
[...]
Collecting werkzeug==2.0.2
  Using cached Werkzeug-2.0.2-py3-none-any.whl (288 kB)
Collecting setuptools<59.7.0,>=59.1.1
ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
    setuptools<59.7.0,>=59.1.1 from https://files.pythonhosted.org/packages/b0/3a/88b210db68e56854d0bcf4b38e165e03be377e13907746f825790f3df5bf/setuptools-59.6.0-py3-none-any.whl#sha256=4ce92f1e1f8f01233ee9952c04f6b81d1e02939d6e1b488428154974a4d0783e (from celery==5.2.3->-r requirements-test-py38.txt (line 39))
[...]
ERROR:   py38: could not install deps [-rrequirements-test-py38.txt]; v = InvocationError('/home/jgangi/Documents/RedHat/exd-sp/cloud-dist/iib/.tox/py38/bin/python -m pip install -rrequirements-test-py38.txt', 1)
```

Reply:
I figured it out, let's see if this build passes

Reply:
From `pip-compile --help` I found this:

```
--allow-unsafe / --no-allow-unsafe
                                  Pin packages considered unsafe: distribute,
                                  pip, setuptools.
                                  
                                  WARNING: Future versions of pip-tools will
                                  enable this behavior by default. Use --no-
                                  allow-unsafe to keep the old behavior. It is
                                  recommended to pass the --allow-unsafe now
                                  to adapt to the upcoming change.

```

Basically the "unsafe" packages are `pip`, `setuptools` and `distribute` but this new version of celery requires it.

I'll upload a new version, removing the comments, but since this behavior will be enabled in future versions it don't seem to be "unsafe" by me.

@yashvardhannanavati @chandwanitulsi @pbortlov what do you think?

Reply:
**Note:** I've sent a new push generated with `--allow-unsafe` and manually edited to remove this line from header after doing `pip-compile`

Reply:
FWIW, I think its better to have the requirements*.txt file be auto generated instead of having that manually changed. If we do want to have it manually edited, how about adding a comment near setuptools indicating the same

Reply:
Good point!

So what should we do next?

1. Auto-generate the file and let it be with `--allow-unsafe`
2. Generate the file and edit manually leaving a comment near `setuptools`
3. Skip this update and let security complain for Celery

As I saw in the `--help` and other sources it's not possible to include `setuptools` on `requirements*` without `--allow-unsafe` by using `pip-compile`